Feature/front register

Hawardshin opened the pull request:
## ✅ 풀\_리퀘스트 체크리스트

<!--
하나씩 확인 후 체크박스에 표시해주세요.
-->

- [x] PR 제목: [Feature/Fix/Refactor...] 작업 내용 한 줄 요약 (브랜치 이름) #이슈번호
- [x] commit message 가 적절한지 확인해주세요.
- [x] 적절한 branch 로 요청했는지 확인해주세요.
- [x] Assignees, Label 을 붙여주세요.
- [x] 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer 로 등록해주세요.
- [x] PR이 승인된 경우 해당 브랜치는 삭제해 주세요!

<br/>

## 🔄 변경 사항

<!-- 해당 pr에서 작업한 내역을 적어주세요. 처음엔 간단하게 요약, list 형식으로 세부사항 작성 -->

Register Page에 캐릭터 선택하는 것을 추가했습니다.

- 이제 캐릭터를 카드로 선택할 수 있습니다.
- 마지막 캐릭터 선택시, 커스텀 캐릭터를 업로드 할 수 있습니다.(미리보기 미구현 + 변경 가능)

<br/>

## 📎 변경한 이유

- 캐릭터를 선택이 아니라 띄워주는 것만 있어서 변경했습니다.

<br/>

<!-- 관련되어있는 Issue Number 를 작성하세요! 해당 이슈를 이곳에 적으면 pr merge 이후 해당 이슈는 자동으로 close 됩니다. -->

closes: #Issue Number

## 👨🏻‍💻 테스트 체크리스트

<!-- 테스트 사항이 있다면 작성해 주세요! -->

- 이러한 테스트를 했어요!

<br/>

## 📌 변경 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->
- 이미지 미리보기가 생각보다 쉬운 건 아닌 것 같습니다. 구현하다가, 머리 깨져서 원상 복구시키고 pull request 날립니다. 나중에 다시 시도해보겠습니다.
<br/>

Hawardshin committed:
Feat: State Array Prototype
- 추후에 백엔드에 보내야하는 state 배열 프로토 타입입니다.!

Hawardshin committed:
Style : Mouse Run Delete
- 마우스 달리기 지웠습니다.
- 추가적으로 현재 사용하지 않는 state는 주석처리함

Hawardshin committed:
Feat: RadioGroup Add
- 여러개의 아바타 선택할 수 있도록 RadioGroup 컴포넌트 추가했습니다.

Hawardshin committed:
Feat: Choose Avatar
- 이제 아바타를 선택할 수 있습니다!

Hawardshin committed:
Feat: Last Avator Click to Upload
- 마지막 아바타를 클릭시 업로드 가능하게 변경했습니다.
- 아바타 미리보기를 구현하려고 했는데 구현하다가 너무 코드가 길어져서
  나중에 state 넣을 때 넣는 방향으로 바꾸는 게 나을 것 같아 일단
  이정도로 하겠습니다.

daejlee commented on frontend/src/components/components/shadcn/radio-group.tsx:
```diff
@@ -0,0 +1,44 @@
+'use client';
+
+import * as React from 'react';
+import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
```
혹시 조신님 환경에서 구동 테스트가 모두 완료되신걸까요?~ ㅜㅜ
- error ./src/components/components/shadcn/radio-group.tsx:4:0
Module not found: Can't resolve '@radix-ui/react-radio-group'
이런 에러가 이 줄에서 뜨네용.. 개인적으로 찾아서 임포트해야하는 사안인가욥?

Hawardshin replied:
앗 혹시 npm install 다시하고 해도 그런가요?

Hawardshin replied:
클러스터 맥에서 구동 테스트는 했습니다!

daejlee replied:
다 삭제하고 클론 받아서 하니 되네용 ㅋㅋㅋ 제 환경이 이상햇던 걸로...

daejlee commented on frontend/src/pages/welcome/register/AvatarContainer.tsx:
```diff
@@ -10,21 +16,46 @@ export default function AvatarContainer() {
     'sloth_health'
   ];
+
+  function uploadHandler(id: string) {
```
15번째 CN 아바타를 누르면 커스텀 아바타를 업로드 할 수 있게끔 하신 것 같군요!!~ 좋네요~ 그러면 밑에 커스텀 아바타 추가하기는 삭제해도 괜춘할듯?

Hawardshin replied:
좋은 생각이에요~~

daejlee commented on frontend/src/pages/welcome/register/AvatarContainer.tsx:
```diff
@@ -10,21 +16,46 @@ export default function AvatarContainer() {
-        rowSquares.push(<AvatarIcon key={idx} avatarName={AvatarList[idx]} />);
+        const id = idx.toString();
+        rowSquares.push(
+          <div className='flex items-center space-x-2'>
```
콘솔창에서 저번에 떴던 경고가 똑같이 뜨네욥ㅜ
55번째 줄 key={id}부분을 이 줄로 갖고와서 적용시키니 경고가 사라졌습니다.
제 생각엔 이 div가 child으로써의 역할을 수행하는 것 같네용

Hawardshin replied:
변경 완료했습니다!!

Hawardshin committed:
Rename : Change Components Directory
- 컴포넌트 폴더구조가 보기 지저분한 부분이 있어서 전체를 변경했습니다.
- 이제 페이지에 해당하는 건 최대한 페이지에만 두고 그 이외의 것들은
  컴포넌트 폴더에 두는 걸로 하면 될 듯 합니다..!

Hawardshin committed:
Feat: Add Dialog Components
- 다이얼로그 버튼을 설정 버튼을 위해 가져왔습니다.

Hawardshin committed:
Fix: Avator Empty Error Edit
- 아바타가 비었을때 발생하는 에러를 없앴고 추후에 미리보기 가능하도록
  값을 넣어줬습니다.

daejlee commented on frontend/.eslintrc.json:
```diff
@@ -23,7 +23,7 @@
   "rules": {
     "import/extensions": [
-      "error",
+      // "error",
```
편안~

daejlee commented on frontend/src/components/avatar/AvatarIcon.tsx:
```diff
@@ -3,7 +3,12 @@ import {Avatar, AvatarFallback, AvatarImage} from '../shadcn/avatar';
     <Avatar>
-      <AvatarImage src={`/character/${avatarName}.svg`} />
+      {avatarName == '' ? (
+        '' // 추후에 이곳에 업로드한 아바타 이미지가 올 수 있도록 업데이트
```
좋아용~~

daejlee reviewed (approved):
고생 많으셨어요~~ LGTM입니닷!
